src/renderer/data/accounts/programChanges.ts

```diff
@@ -55,20 +55,26 @@ export const subscribeProgramChanges = async (
     const solConn = new sol.Connection(netToURL(net));
     const subscriptionID = solConn.onProgramAccountChange(
       programIDPubkey,
-      (info: sol.KeyedAccountInfo /* , ctx: sol.Context */) => {
+      (info: sol.KeyedAccountInfo, ctx: sol.Context) => {
+        if (setValidatorSlot) {
+          setValidatorSlot(ctx.slot);
```

Comment:
Is this going to cause a lot of re-renders? This method will be called a lot of times, so I think `setValidatorSlot` will be called on every single program change, and I'm not sure React is smart enough to batch up the updates...

Reply:
might be better off pulled into the https://github.com/workbenchapp/solana-workbench/pull/86/files#diff-fb0f916c5d552a20f81d5b73c0ac98e782455d3d2c0457d0ea2c514f90cfe025R98 block

Reply:
after sleeping on this, i think i'm going about this wrong because i'm refactoring what was there before.

ideally, programchanges should just be one of many sources of account info in the cache, and then "what's the top X max delta accounts" should also be a query from the cache...

the point of the account info cache being to abstract the multiple sources of account info (and their age) from the multiple uses (calculation, rendering, whatever) of account info.

Reply:
oh, and no, starred accounts and programchange (account source) shouldn't know about each other - that's why `getAccounts()` uses a shared cache of account info.